contrib/!window-management/perspectives/packages.el

```diff
@@ -1,107 +1,457 @@
+;;
+;; Copyright (c) 2012-2014 Sylvain Benner
+;; Copyright (c) 2014-2015 Sylvain Benner & Contributors
+;;
+;; Author: Sylvain Benner <sylvain.benner@gmail.com>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
 (setq perspectives-packages
-  '(
-    perspective
-    persp-projectile
-    ))
+      '(
+        ;; perspective
+        persp-mode
+        ;; persp-projectile
+        ))
+
+(defun perspectives/init-persp-mode ()
+  (use-package persp-mode
+    :no-require t
+    :init
+    :config
+    ;; switch off the animation of restoring window configuration
+    (persp-mode 1)
+
+    ;; This gives reverse order for how they were added
+    ;; (persp-names)
+    ;; This lists them alphabetically
+    ;; (persp-names-sorted)
+
+    (defvar persp-switch-wrap t
+      "Whether `persp-next' and `persp-prev' should wrap.")
+
+    (defun persp-curr-name ()
+      (interactive)
+      (safe-persp-name (get-frame-persp)))
+
+    (defun persp-next ()
+      "Switch to next perspective (to the right)."
+      (interactive)
+      (let* ((names (reverse (persp-names)))
+             (pos (cl-position (persp-curr-name) names)))
+        (cond
+         ((null pos) (persp-switch persp-nil-name))
+         ((= pos (1- (length names)))
+          (if persp-switch-wrap (persp-switch (nth 0 names))))
+         (t (persp-switch (nth (1+ pos) names))))))
+
+    (defun persp-prev ()
+      "Switch to previous perspective (to the left)."
+      (interactive)
+      (let* ((names (reverse (persp-names)))
+             (pos (cl-position (persp-curr-name) names)))
+        (cond
+         ((null pos) (persp-switch persp-nil-name))
+         ((= pos 0)
+          (if persp-switch-wrap (persp-switch (nth (1- (length names)) names))))
+         (t (persp-switch (nth (1- pos) names))))))
+
+    (defface persp-selected-face
+      '((t (:inherit font-lock-keyword-face)))
+      "The face used to highlight the current perspective on the modeline.")
+
+    (defun persp-format-name (name)
+      "Format the perspective name given by NAME for display in `persp-modestring'."
+      (let ((string-name (format "%s" name)))
+        (if (equal name (persp-curr-name))
+            (propertize string-name 'face 'persp-selected-face)
+          string-name)))
+
+    (defun spacemacs//perspectives-ms-documentation ()
+      "Return the docstring for the perspectives micro-state."
+      (let* ((persp-list           (reverse (persp-names)))
+             (formatted-persp-list (mapconcat
+                                    (lambda (persp) (persp-format-name persp))
+                                    persp-list " ")))
+        (concat formatted-persp-list
+                (when perspective-display-help
+                  (concat
+                   ;; "\n[n] next, [p/N] previous, [TAB] back and forth, "
+                   "\n"
+                   "[n|p] [next|previous] perspective, "
```

Comment:
The `[next|previous]` is highlighted the same way as keys which is inconsistent with the other micro-states in Spacemacs. I think this should be `[n/p] next|previous perspective` and that style should be applied for the rest of the micro-state docs. I also think that each key and description pair could be on their own lines to be more readable. The window micro-state has a longer docstring and does it this way.